test/jdk/javax/swing/JToggleButton/bug4128979.java

```diff
@@ -22,63 +22,65 @@
  */
 
 /* @test
-   @bug 4128979
-   @requires (os.family == "windows")
-   @summary Tests that background changes correctly in WinLF for JToggleButton when pressed
-   @key headful
-   @run applet/manual=yesno bug4128979.html
-*/
+ * @bug 4128979
+ * @requires (os.family == "windows")
+ * @modules java.desktop/com.sun.java.swing.plaf.windows
+ * @library /java/awt/regtesthelpers
+ * @build PassFailJFrame
+ * @summary Tests that background changes correctly in WinLF for JToggleButton when pressed
+ * @run main bug4128979
+ */
+
+import java.awt.Color;
+import java.awt.Component;
+import java.awt.ComponentOrientation;
+import java.awt.Container;
+import java.awt.FlowLayout;
+import java.awt.Graphics;
 
 import javax.swing.BorderFactory;
 import javax.swing.BoxLayout;
 import javax.swing.Icon;
-import javax.swing.JApplet;
 import javax.swing.JFrame;
 import javax.swing.JLabel;
-import javax.swing.JOptionPane;
 import javax.swing.JPanel;
 import javax.swing.JToggleButton;
 import javax.swing.JToolBar;
+import javax.swing.SwingUtilities;
 import javax.swing.UIManager;
-import javax.swing.UnsupportedLookAndFeelException;
 import javax.swing.WindowConstants;
-import java.awt.BorderLayout;
-import java.awt.Color;
-import java.awt.Component;
-import java.awt.ComponentOrientation;
-import java.awt.Container;
-import java.awt.FlowLayout;
-import java.awt.Graphics;
 
-public class bug4128979 extends JApplet {
-
-    public static void main(String[] args) {
-        JApplet applet = new bug4128979();
-        applet.init();
-        applet.start();
-
-        JFrame frame = new JFrame("Test window");
-        frame.setDefaultCloseOperation(WindowConstants.EXIT_ON_CLOSE);
-        frame.setLayout(new BorderLayout());
-        frame.add(applet, BorderLayout.CENTER);
-        frame.setSize(600, 240);
-        frame.setVisible(true);
+import com.sun.java.swing.plaf.windows.WindowsLookAndFeel;
+
+public class bug4128979 {
+    private static JFrame frame;
+    private static final String INSTRUCTIONS = """
+            When the test starts, toggle buttons are visible in three rows
+            two of which are toolbars.
+
+            Press these buttons, their background color must change
+            to half tones between the button background colors and the ToggleButton
+            highlight color (it is shown in the square below).
+
+            If the background color does not change correctly for at least one button,
+            the test fails. """;
+
+    public static void main(String[] args) throws Exception {
+        UIManager.setLookAndFeel("com.sun.java.swing.plaf.windows.WindowsClassicLookAndFeel");
```

Comment:
As the test is for Windows only it is good to have OS check along with the `@requires (os.family == "windows")` and throw skipped exception otherwise.
Since it is followed by setting `WindowsClassicLookAndFeel` all the more reason to include this check.

```suggestion
         if (OSInfo.getOSType() != OSInfo.OSType.WINDOWS) {
            throw new SkippedException("This test is for Windows only");
        }
        UIManager.setLookAndFeel("com.sun.java.swing.plaf.windows.WindowsClassicLookAndFeel");
```

Reply:
Updated.

Reply:
You can remove the OS check here if you would like to, since it is not strictly required when run as jtreg test (which is essentially the case here).

Reply:
I will keep this check as of now.